ballerina/tests/tests.bal

```diff
@@ -0,0 +1,359 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/oauth2;
+import ballerina/test;
+
+configurable string clientId = ?;
+configurable string clientSecret = ?;
+configurable string refreshToken = ?;
+
+// test discount ids for batch and basic endpoints.
+string discount_id = "";
+string[] batch_discount_ids = [];
+
+// discount properties for basic create
+string hs_value = "";
+string hs_label = "";
+string hs_type = "";
+
+// discount properties for update
+string new_hs_value = "8";
+string new_hs_label = "test_updated_label";
+string new_hs_type = "PERCENT";
+
+ConnectionConfig config = {
+    auth: {
+        clientId,
+        clientSecret,
+        refreshToken,
+        credentialBearer: oauth2:POST_BODY_BEARER
+    }
+};
+
+final Client hubspotClient = check new (config);
+
+@test:Config {
+    dependsOn: [testBatchCreate]
+}
+function testList() returns error? {
+
+    GetCrmV3ObjectsDiscountsQueries params = {
+        'limit: 10,
+        archived: false,
+        properties: ["hs_label", "hs_value", "hs_type"]
+    };
+
+    CollectionResponseSimplePublicObjectWithAssociationsForwardPaging|error response = check hubspotClient->/.get({}, params);
+    if response is CollectionResponseSimplePublicObjectWithAssociationsForwardPaging {
+        test:assertNotEquals(response.results, [], "No discounts found");
+        test:assertTrue(response.results.length() <= 10, "Limit Exceeded");
+        test:assertNotEquals(response.results[0].id, (), "Discount id is not found");
+        test:assertNotEquals(response.results[0].properties, (), "Discount properties are not found");
+        test:assertNotEquals(response.results[0].properties["hs_type"], (), "Discount label is not found");
+        test:assertNotEquals(response.results[0].properties["hs_value"], (), "Discount value is not found");
+        test:assertNotEquals(response.results[0].properties["hs_label"], (), "Discount type is not found");
```

Comment:
Instead of individually checking and validating these values, can't we compare the entire response with an expected response? if that's possible let's do that. And update the other test cases tool